src/%IPM/ExtensionBase/ServerMenu/Interface.cls

```diff
@@ -0,0 +1,25 @@
+/// Additional inteface for server menu source control extensions to implement.
+Class %IPM.ExtensionBase.ServerMenu.Interface Extends %Studio.Extension.Base [ Abstract ]
```

Comment:
Why we still add Studio support?

Reply:
Good question! This is the abstract super class for %IPM.Extension.ServerMenu.MenuItems to be used during init, in order to avoid invalid references. Itself is just an abstract class but our studio ui extension, or user defined ui extension, can extend this class. So we are not adding Studio support, but just provide users with super class to add their studio support.

Reply:
@isc-tleavitt  This class is to 1: provide a dummy super class for user-defined server menu extension class + 2: avoid invalid references to server menu extension in IPM. For example, when initializing IPM, if user do not have a command line prompt for server menu extension, we should query for classes that extend this interface class. 
Based on the query result, if there is exactly one match, we set that as the extension using the composite method. 
If more than one match, and if not in quiet mode, we let the user select which one to set as the extension in the Shell interface.
If no match, do not set.
So that this behaves similar to the current behave of PM extension. Does it make sense?

Reply:
@Beiming-Zhang why not just query for non-abstract subclasses of %Studio.Extension.Base? Also, mere existence of such a class is not sufficient to indicate that it should be activated as the source control class for the namespace. We just want to overlay ##class(%Studio.SourceControl.Interface).SourceControlClassGet() (if non-empty) with the PM extension on init.

Reply:
@isc-tleavitt  We can query on %Studio.Extension.Base, however we will then need to split out other source-control classes such as Perforce.Main which I think would be easier to just separate out the interface for SourceControl extension(%IPM.ExtensionBase.SourceControl.Interface) vs. ServerMenu extension(%IPM.ExtensionBase.ServerMenu.Interface).